framework/src/systems/NonlinearSystemBase.C

```diff
@@ -1456,6 +1460,12 @@ NonlinearSystemBase::computeResidualInternal(const std::set<TagID> & tags)
   }
   PARALLEL_CATCH;
 
+  if (_computing_scaling_residual)
+    // We computed the volumetric objects. We can return now (after assembling) before we get into
+    // any strongly enforced constraint conditions or penalty-type objects
+    // (DGKernels, IntegratedBCs, InterfaceKernels, Constraints)
+    return;
+
```

Comment:
DG simply won't work with scaling?

Reply:
DG can sometimes use penalty factors which is bad for determining a proper scaling factor. Moreover, except if a user is doing DG with pure advection (no diffusion, no sources, no sinks) and constant monomial basis functions, they are going to have regular Kernels in addition to their DGKernels. This is because there are elemental contributions in addition to the face contributions.